apiserver/common/unitstate.go

```diff
@@ -51,8 +51,12 @@ func (s UnitStateState) Unit(name string) (UnitStateUnit, error) {
 // UnitStateService describes the ability to retrieve and persist
 // remote state for informing hook reconciliation.
 type UnitStateService interface {
+	// GetUnitUUIDForName returns the UUID corresponding to the input unit name.
+	GetUnitUUIDForName(ctx context.Context, name string) (string, error)
 	// SetState persists the input agent state.
 	SetState(context.Context, unitstate.AgentState) error
+	// GetState returns the full agent state of the unit.
```

Comment:
Unit state is not the state of the agent.  I'm very confused by the structure being called AgentState as well. 

```suggestion
	// GetState returns the internal state of the unit. The return data will be empty if no hook has been run for this unit. 
```

I have another comment on this topic.

Reply:
As far as I understand it it is called agent state because it contains more than just the uniter state. I.e. it contains the charm, secret, storage and relation state as well. I guess this is what makes up the agent state?

Reply:
The uniter doesn't know about the unit agent. The data here is the uniter's knowledge of what its own state is. Has it run Install? has it previously configured an relations? What hook it is in the process of running or not. The data the uniter needs at startup to not send a relation_join hook for an existing relation. The data the uniter needs not to run the install hook a second time if it's been restarted due to agent upgrade.

Reply:
Added comment and changed the struct names to UnitAgentState